library/ssl_tls.c

```diff
@@ -2042,6 +2074,28 @@ static int ssl_tls13_session_load( mbedtls_ssl_session *session,
     memcpy( session->resumption_key, p, session->resumption_key_len );
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && defined(MBEDTLS_SSL_CLI_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        /* load host name */
+        if( end - p < 1 )
+            return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+        session->hostname_len = p[0];
+        p += 1;
```

Comment:
align with `opaque hostname<1..2^16-1>;` at line 1929
```suggestion
        if( end - p < 2 )
            return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
        session->hostname_len = MBEDTLS_SSL_GET_UINT16_BE( p, 0);
        p += 2;
```